lib/scenic/adapters/postgres/index_reapplication.rb

```diff
@@ -0,0 +1,63 @@
+module Scenic
+  module Adapters
+    class Postgres
+      # Dropping a materialized view causes all indexes on the materialized view
+      # to be dropped. This object can be used to capture the existing
+      # materialized views before updating the materialized view and then
+      # reapplying after.
+      #
+      # @api private
+      class IndexReapplication
+        def initialize(connection:, speaker: ActiveRecord::Migration)
+          @connection = connection
+          @speaker = speaker
+        end
+
+        # Caches indexes on the provided object before executing the block and
+        # then reapplying the indexes. Errors in applying the indexes are caught
+        # and logged to output.
+        #
+        # @param on The name of the object we are reapplying indexes on.
+        # @yield Operations to perform before reapplying indexes. @return [void]
+        def on(name)
+          indexes = Indexes.new(connection: connection).on(name)
+
+          yield
+
+          indexes.each(&method(:try_index_create))
```

Comment:
*really* trying to make `&method` a thing... I wish ruby would happily let me drop some of those parens, but it won't. I still like it, I think.